tephra-core/src/main/java/org/apache/tephra/persist/AbstractTransactionLog.java

```diff
@@ -48,21 +52,30 @@
   protected long timestamp;
   private volatile boolean initialized;
   private volatile boolean closed;
+  private AtomicLong writtenUpTo = new AtomicLong();
```

Comment:
I think this can also be just a `long`.

Reply:
yes it is only called within synchronized {}